js/models/site.es6.js

```diff
@@ -40,9 +40,25 @@ Site.prototype = $.extend({},
           let tab = backgroundPage.tabs[this.tabId];
           let host = backgroundPage.utils.extractHostFromURL(tab.url);
           let site = backgroundPage.Sites.get(host);
-          if(site){
+
+          // Determine if this is a special page, eg extensions, and reset domain and disabled flag.
+          
+          this.disabled = true;
+
+          if (site) {
               this.site = site;
               this.isWhitelisted = site.whiteListed;
+              
+              let special = site.specialDomain();
+              if (special) {
+                  this.domain = special;    // eg "extensions", "options", "new tab"
+              }
+              else {
+                  this.disabled = false;
+              }
+          }
+          else {
+              this.domain = '-';    // should not happen
```

Comment:
For readability, can we flip these if/else so this is more readable. So the short one is at the top -- 
`if (!site) { this.domain = '-'; } else { ... }`